.github/workflows/checkAndSubmitAddonMetadata.yml

```
@@ -179,6 +179,137 @@ jobs:
       uses: peter-evans/close-issue@v3
       with:
         issue-number: ${{ inputs.issueNumber }}
+  createReviewComment:
+    # jq for windows has issues parsing multiline strings (e.g. CRLF),
+    # use linux instead.
+    runs-on: ubuntu-latest
+    needs: [checkMetadata]
+    strategy:
+      matrix:
+        python-version: [ 3.11 ]
+    permissions:
+      contents: write
+      discussions: write
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      ref: ${{ inputs.headRef }}
+    - name: Get add-on name and version
+      id: getAddonNameAndVersion
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const addonFilename = ${{ needs.getAddonFilename.outputs.addonFileName }}
+          const addonName = addonFilename.split('/')[1]
+          core.setOutput('addonName', addonName)
+          const addonVersion = addonFilename.split('/')[2].replace('.json', '')
+          core.setOutput('addonVersion', addonVersion)
+    - name: Check if discussion for this add-on exists
+      id: checkDiscussion
+      run: |
+        addonId=$(
+          echo ${{ needs.getAddonFilename.outputs.addonFileName }} \
+          | sed -r "s|addons/(.*)/.*\.json|\1|g"
+        )
+        jqCode="
+        . | .[\"$addonId\"]
+        "
+        jq -e "$jqCode" discussions.json
+        # -e only sets the exit status of jq to 0 if there is "truthy" output.
+        # If no valid result is produced, i.e. no add-on id is found, jq exits with status 4.
+        # If there is a different error, a different non-zero exit code is used.
+        exit $? # Exit with the same exit code as jq
+
+    - name: Create discussion for add-on id
+      id: createDiscussion
+      if: ${{ steps.checkDiscussion.outcome }} == 'failure'
+      uses: abirismyname/create-discussion@v1.1.0
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}      
+      with:
+        repository-id: $GITHUB_REPOSITORY_ID
```

I think this should be

```suggestion
        repository-id: ${{ github.repository_id }}
```

---

@nvdaes - does [this method](https://stackoverflow.com/a/72834015/8030743) work here?

Something similar might also work for fetching the discussion category

---

> no, as I mentioned, the repository id needs to be hard-coded since the variable suggested by you gives an id that is different respect that the id we need here.

@nvdaes - did you test the linked method? for me the query

For me query below yielded the correct result `'MDEwOlJlcG9zaXRvcnkyMzI4MTU5NTQ=`

```
query {
  repository (name: "addon-datastore", owner: "nvaccess")  {
        id
  }
}
```

---

This query also fetches the discussion category ID, if the discussion is the first discussion

```
query {
  repository (name: "addon-datastore", owner: "nvaccess")  {
        id
        discussionCategories(first: 1) {
      		edges {
      		  node {
      		    id
      		  }
      		}
      		nodes {
      		  id
      		}
    		}
  }
}
```

---

you can replace the repo name and owner with the github env variables so this is truly independent of the repository environment, as long as the target discussion category is the first discussion category.